### PR TITLE
chore: update update-data-dependencies.sh script to allow for bumping to a tag version

### PR DESCRIPTION
**Pull request by alharris-at**
#### Description of changes
I made these changes to make supporting the `feature/rds-support` branch a bit easier. Change is already present there.

#### Issue #, if available
N/A

#### Description of how you validated changes
Tested on the `feature/rds-support` branch where the package.json file specifies the tag to use.

https://github.com/aws-amplify/amplify-cli/blob/feature/rds-support/package.json#L49

#### Checklist
- [x] PR description included

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by sobolk — APPROVED** [review]

File: scripts/update-data-dependencies.sh [hunk 1]
```diff
@@ -1,6 +1,23 @@
...
+        --deep \
+        --upgrade \
+        --filter "$FILTER"
+  else 
+    echo "Updating to $1 tag"
+    npx ncu \
+        --deep \
+        --upgrade \
+        --filter "$FILTER" \
+        --target $1
```
Comment (sobolk) [inline scripts/update-data-dependencies.sh:22]: @alharris-at 
thing to consider. This doesn't update peer deps by default. should we include them here?

Comment (alharris-at) [reply scripts/update-data-dependencies.sh:22]: Yeah, I'll update to specify the peer deps as well. I don't think it'll make an impact in CLI, but we want that in the codegen/api category repos for sure, and I'd like to keep the functionality consistent.

**Review by sobolk — APPROVED** [review]

File: scripts/update-data-dependencies.sh [hunk 2]
```diff
@@ -1,6 +1,25 @@
...
+#      `npx scripts/update-data-dependencies.sh update-data-packages @rds-support`
+
+FILTER="amplify-codegen @aws-amplify/amplify-category-api @aws-amplify/graphql-auth-transformer @aws-amplify/graphql-default-value-transformer @aws-amplify/graphql-function-transformer @aws-amplify/graphql-http-transformer @aws-amplify/graphql-index-transformer @aws-amplify/graphql-maps-to-transformer @aws-amplify/graphql-model-transformer @aws-amplify/graphql-predictions-transformer @aws-amplify/graphql-relational-transformer @aws-amplify/graphql-schema-test-library @aws-amplify/graphql-searchable-transformer @aws-amplify/graphql-transformer-core @aws-amplify/graphql-transformer-interfaces @aws-amplify/graphql-transformer-migrator graphql-auth-transformer graphql-connection-transformer graphql-dynamodb-transformer graphql-elasticsearch-transformer graphql-function-transformer graphql-http-transformer graphql-key-transformer graphql-mapping-template graphql-predictions-transformer graphql-relational-schema-transformer graphql-transformer-common graphql-transformer-core graphql-transformers-e2e-tests graphql-versioned-transformer"
```
Comment (AaronZyLee) [inline scripts/update-data-dependencies.sh:7]: Just one question. The filter also includes `amplify-codegen` which might not have a tagged release at the same time with API category. Will this script ignore the update if the tag does not exist?

**Review by AaronZyLee — APPROVED** [review]

File: scripts/update-data-dependencies.sh (hunk 2 above)
Comment (alharris-at) [reply scripts/update-data-dependencies.sh:7]: Yep! It'll only pull if there's a relevant update.

**Pull request merged**